blocks/subscriptions-block/features/checkout/default.jsx

```diff
@@ -1,118 +1,279 @@
+	const { loginURL, offerURL, stripeIntentsID, successURL } = customFields;
 
-	const [loggedIn, setIsLoggedIn] = useState(false);
-	const [user, setUser] = useState(false);
-	const [signedInIdentity, setSignedInIdentity] = useState(false);
-	const [showPaymentScreen, setShowPaymentScreen] = useState(false);
-	const [userInfo, setUserInfo] = useState({});
+	const initialState = {
+		account: true,
+		billingAddress: false,
+		payment: false,
+		review: false,
+	};
 
-	const { Identity, getSignedInIdentity } = useIdentity();
+	const { Sales } = useSales();
+	const { Identity } = useIdentity();
 	const phrases = usePhrases();
 
-	const [isInitialized, setIsInitialized] = useState(false);
+	const {
+		stripeIntents,
+		paypal,
+		error: errorPaymentOptions,
+		isFetching: isFetchingPaymentOptions,
+	} = usePaymentOptions(stripeIntentsID);
 
 	const params = new URLSearchParams(window.location.search);
-	const token = params.get("token");
+	const paypalToken = params.get("token");
 
-	useEffect(() => {
-		const isOrderNumberInLocalStorage = localStorage.getItem(LABEL_ORDER_NUMBER_PAYPAL);
-		if (isOrderNumberInLocalStorage && token) {
-			setIsInitialized(true);
-		}
-	}, []);
+	const { error: errorPaypal, isCheckingPaypal } = usePayPalPaymentRedirect(paypal, paypalToken, successURL);
 
-	useEffect(() => {
-		const isLoggedIn = async () => {
-			setIsLoggedIn(await Identity.isLoggedIn());
-		};
+	const { isFetching, isFetchingCartOrder, isLoggedIn, cartDetail, orderDetail } =
+		useCartOrderDetail(paypalToken);
+
+	const [user, setUser] = useState(false);
+	const [isOpen, setIsOpen] = useState(initialState);
+	const [isComplete, setIsComplete] = useState(initialState);
+
+	const [billingAddress, setBillingAddress] = useState({});
+	const [cart, setCart] = useState();
+
+	const [captchaToken, setCaptchaToken] = useState();
+	const [captchaError, setCaptchaError] = useState();
+	const [resetRecaptcha, setResetRecaptcha] = useState(true);
+
+	const [error, setError] = useState();
 
-		isLoggedIn();
-	}, [Identity]);
+	const [order, setOrder] = useState();
+	const { isRecaptchaEnabled } = useRecaptcha("checkout");
+
+	const checkoutURL = window.location.href;
 
 	useEffect(() => {
 		let isActive = true;
 
-		if (loggedIn) {
+		if (isLoggedIn) {
 			Identity.getUserProfile()
 				.then((userProfile) => {
 					if (isActive) {
 						setUser(userProfile);
-						setSignedInIdentity(getSignedInIdentity(userProfile));
+						setIsOpen((state) => ({ ...state, account: false, billingAddress: true }));
+						setIsComplete((state) => ({ ...state, account: true }));
 					}
 				})
 				.catch(() => {
 					setUser(false);
 				});
 		}
 
-		// cancel subscription to useEffect
 		return () => {
 			isActive = false;
 			return null;
 		};
-	}, [Identity, loggedIn]);
+	}, [Identity, isLoggedIn]);
+
+	useEffect(() => {
+		if (!isFetching) {
+			if (!isLoggedIn && loginURL && checkoutURL) {
+				window.location.href = `${loginURL}?redirect=${checkoutURL}`;
```

Comment:
What if someone passes in a `loginURL` of `http://my.domain/login?foo=bar`?  We should be using the URL API.

```
const loginURLObject = new URL(loginURL);
loginURLObject.searchParams.append('redirect', checkoutURL);
window.location.href = loginURLObject.toString();
```
This will also encode the url if the user tries something:
![image](https://github.com/WPMedia/arc-themes-blocks/assets/2287238/c3b3c770-739f-4681-8493-9ac02a6b2626)

Reply:
@nschubach I think we should handle this update in a different ticket/PR, since across different blocks/files we are just using `window.location.href = "..."`. Thus we can create a function, check if it works under all scenarios and update it accordingly 
<img width="472" alt="Screenshot 2024-05-08 at 7 21 45 a m" src="https://github.com/WPMedia/arc-themes-blocks/assets/54566275/4a63f4f2-6895-403e-90a7-96ab7437ed51">

Reply:
I can pass this for now, I agree it's not something we've paid attention to in the past and it's likely a controlled situation here.